site/content/how-to/traffic-management/response-headers.md

```diff
@@ -0,0 +1,209 @@
+---
+title: "HTTP Response Headers"
+description: "Learn how to modify the response headers of your application using NGINX Gateway Fabric."
+weight: 700
+toc: true
+docs: ""
+---
+
+[HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/) filters can modify the headers during the request-response lifecycle. [HTTP Header Modifiers](https://gateway-api.sigs.k8s.io/guides/http-header-modifier/?h=request#http-header-modifiers) can be used to add, modify or remove headers in incoming requests.
+
+1. [ResponseHeaderModifier](https://gateway-api.sigs.k8s.io/guides/http-header-modifier/?h=request#http-response-header-modifier) to alter headers in response before responding to the downstream.
+
+
+In this guide we will modify the headers for HTTP responses when client requests are made. For an introduction to exposing your application, we recommend that you follow the [basic guide]({{< relref "/how-to/traffic-management/routing-traffic-to-your-app.md" >}}) first.
+
+
+## Prerequisites
+
+- [Install]({{< relref "/installation/" >}}) NGINX Gateway Fabric.
+- [Expose NGINX Gateway Fabric]({{< relref "installation/expose-nginx-gateway-fabric.md" >}}) and save the public IP
+  address and port of NGINX Gateway Fabric into shell variables:
+
+   ```text
+   GW_IP=XXX.YYY.ZZZ.III
+   GW_PORT=<port number>
+   ```
+
+{{< note >}}In a production environment, you should have a DNS record for the external IP address that is exposed, and it should refer to the hostname that the gateway will forward for.{{< /note >}}
+
+
+## Response Header Filter
+
+In this guide, we will configure an HTTPRoute with a `ResponseHeaderModifier` filter to modify the HTTP response headers of an application.
+
+
+### Deploy the Headers application
+
+Begin by deploying the example application `headers`:
+
+   ```shell
+   kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.3.0/examples/http-response-header-filter/headers.yaml
+   ```
+
+Verify if the Pod is running in the `default` Namespace:
+
+   ```shell
+   kubectl -n default get pods
+   ```
+
+   ```text
+   NAME                      READY   STATUS    RESTARTS   AGE
+   headers-6f854c478-k9z2f   1/1     Running   0          32m
+   ```
+
+### Deploy the Gateway API Resources for the Header Application
+
+
+The [gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) resource is typically deployed by the [cluster operator](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the gateway:
```

Comment:
@ADubhlaoich do we just need to take initiative and go through and capitalize all resource names (like Gateway) as we had talked about months ago? I wasn't sure if the docs team had an action item or not.

Reply:
The capitalization of the resource names is inconsistent in this doc. Can we choose one style and stick with it?

Reply:
Didn't noticed I'd been tagged in a thread. Consistency's the most important: capitalise any resource names that haven't been. We have better defined rules about this now but I will fix any issues myself at a later point.